### PR TITLE
Refactor/user endpoints

### PR DESCRIPTION
**Pull request by ruihildt**
# Description

Update endpoints and corresponding tests.

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

File: __test__/mentor.test.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ describe('Get /mentor/:username/mentors', () => {
     const jwtToken = await jwt.generateToken(user);
     return request(server)
       .get('/api/mentor/john/mentors')
-      .set({ token: jwtToken })
+      .set({ authorization: jwtToken })
```
Comment (iamcristos) [inline __test__/mentor.test.js]: is there a reason you changing the token to authorization????

**Comment by ruihildt** [comment]
@iamcristos Please note this PR is not ready for review.

File: __test__/mentor.test.js (hunk 1 above)
Comment (iykekings) [reply __test__/mentor.test.js]: Yeah, the token is set as authorization header

**Comment by ruihildt** [comment]
We won't have time to evaluate this possible change before the end of Labs.

**Pull request closed without merging**